src/ProjectService.Validation/EditProjectValidator.cs

```diff
@@ -1,105 +1,155 @@
+
+              RuleFor(x => (string)GetOperationByPath(x, ShortName).value)
+                .NotEmpty()
+                .MaximumLength(30)
+                .WithMessage("Short name is too long.");
+            });
+
+          When(x => GetOperationByPath(x, Description) != null, () =>
+            {
+              RuleFor(x => x.Operations).UniqueOperationWithAllowedOp(Description, "replace");
+            });
+
+          When(x => GetOperationByPath(x, ShortDescription) != null, () =>
+            {
+              RuleFor(x => x.Operations).UniqueOperationWithAllowedOp(ShortDescription, "replace");
+
+              RuleFor(x => (string)GetOperationByPath(x, ShortDescription).value)
+                .MaximumLength(300)
+                .WithMessage("Short description is to long");
+            });
+
+          When(x => GetOperationByPath(x, Status) != null, () =>
             {
-                Name,
-                ShortName,
-                Description,
-                ShortDescription,
-                Status,
-                DepartmentId
-            };
-
-        public static string Name => $"/{nameof(EditProjectRequest.Name)}";
-        public static string ShortName => $"/{nameof(EditProjectRequest.ShortName)}";
-        public static string Description => $"/{nameof(EditProjectRequest.Description)}";
-        public static string ShortDescription => $"/{nameof(EditProjectRequest.ShortDescription)}";
-        public static string Status => $"/{nameof(EditProjectRequest.Status)}";
-        public static string DepartmentId => $"/{nameof(EditProjectRequest.DepartmentId)}";
-        Func<JsonPatchDocument<EditProjectRequest>, string, Operation> GetOperationByPath =>
-            (x, path) =>
-                x.Operations.FirstOrDefault(x =>
-                    string.Equals(x.path, path, StringComparison.OrdinalIgnoreCase));
-
-        public EditProjectValidator()
+              RuleFor(x => x.Operations).UniqueOperationWithAllowedOp(Status, "replace");
+
+              RuleFor(x => (ProjectStatusType)GetOperationByPath(x, Status).value)
+                .IsInEnum()
+                .WithMessage("Wrong status value.");
+            });
+
+          When(x => GetOperationByPath(x, DepartmentId) != null, () =>
+            {
+              RuleFor(x => x.Operations).UniqueOperationWithAllowedOp(DepartmentId, "replace");
+
+              RuleFor(x => GetOperationByPath(x, DepartmentId).value)
+                .Must(x => Guid.TryParse(x.ToString(), out Guid _))
+                .WithMessage("Wrong department id value.")
+                .Must(x => GetDepartment(Guid.Parse(DepartmentId)))
+                .WithMessage("Project department does not found.");
+            });
+        });
+    }
+
+    private bool GetDepartment(Guid? departmentId)
+    {
+      string errorMessage = "Cannot edit project. Please try again later.";
+
+      if (!departmentId.HasValue)
+      {
+        return true;
```

Comment:
its ok, but i think this check should not be in this func. we need check if departamentId has not value before call this func.

Reply:
done